src/components.js

```diff
@@ -811,16 +811,18 @@ const components = [
     style: props => ({
       position: 'absolute',
       top: props.top ? 0 : null,
-      right: props.right ? 0 : null,
+      right: props.center ? null : (props.right ? 0 : null),
       bottom: props.bottom ? 0 : null,
-      left: props.left ? 0 : null,
+      left: props.center ? '50%' : (props.left ? 0 : null),
```

Comment:
I'd suggest pulling these out into smaller functions and adding some tests since it's doing more than just a single ternary (though I haven't gotten to writing many tests yet)